Dockerfile

```diff
@@ -1,18 +1,44 @@
-FROM python:3.11.8
+################################################################################
+# getml-base
+FROM python:3.11.8 AS getml-base
 
 RUN useradd getml
 USER getml
 WORKDIR /home/getml
 
 COPY --chown=getml:getml --chmod=0777 ./requirements.txt /home/getml/requirements.txt
 
-ENV PATH="$PATH:/home/getml/.local/bin"
+ENV PATH="/home/getml/.local/bin:$PATH"
+
 RUN python3.11 \
     -mpip install \
     -r /home/getml/requirements.txt
 
-RUN mkdir /home/getml/.getML /home/getml/.getML/logs /home/getml/.getML/projects /home/getml/demo
-RUN curl https://storage.googleapis.com/static.getml.com/download/1.4.0/getml-1.4.0-x64-linux.tar.gz | tar -C /home/getml/.getML -xvzf -
+RUN mkdir /home/getml/.getML
+
+################################################################################
+# getml-amd64
+FROM getml-base AS getml-amd64
+ARG GETML_ARCH=x64
+
+################################################################################
+# getml-arm64
+FROM getml-base AS getml-arm64
+ARG GETML_ARCH=arm64
+
+################################################################################
+# getml-demo
+FROM getml-${TARGETARCH} AS getml-demo
+
+ARG TARGETOS
+ARG GETML_VERSION
+ARG GETML_BUCKET="https://storage.googleapis.com/static.getml.com/download"
+ARG GETML_ENGINE_FILE="getml-${GETML_VERSION}-${GETML_ARCH}-${TARGETOS}.tar.gz"
```

Comment:
The use of `GETML_VERSION` is misleading. We are hardcoding the getml version to 1.4.0 in `requirements.txt`. The python API is neither forwards nor backwards compatible.

Reply:
:thinking: 
Probably better then to grep the version from the requirements then.

Reply:
Using
`    export GETML_VERSION=$(grep -o "^getml==.*$" requirements.txt | cut -b8-) ;\`